src/ray/core_worker/core_worker.cc

```diff
@@ -810,9 +810,11 @@ Status CoreWorker::AllocateReturnObjects(
 }
 
 Status CoreWorker::ExecuteTask(const TaskSpecification &task_spec,
-                               const ResourceMappingType &resource_ids,
+                               const std::shared_ptr<ResourceMappingType> resource_ids,
                                std::vector<std::shared_ptr<RayObject>> *return_objects) {
-  resource_ids_ = resource_ids;
+  if (resource_ids != nullptr) {
+    resource_ids_ = *resource_ids;
```

Comment:
```suggestion
    resource_ids_ = resource_ids;
```
Make resource_ids_ a shared_ptr to avoid copy? Or does this not make sense.